### ISSUE TITLE
xgboost doesn't balance on data in R

### ISSUE THREAD

**Issue by Frankyboy1994** [dmlc/xgboost#2702]
Hello everybody,

I'm currently working on a xgboost model for damage insurance individual reserving. The goal is to predict the final cost for each claim, using the most recent information. Therefore, it is a regression problem.

For this problem, I'm using the xgb.train funciton of the xgboost package in R. The model predicts well the final cost, but the model doesn't balance. What I mean by this is that the sum of predicted values is far away from the sum of observed values. I would like the 2 sums to be closer. In fact, the model predicts well the small values of observed, and wrong the high values of observed.

I tried to put more weight on high observed values, but it didn't work very well. 

Is there a parameter in the xgb.train function which I can tune in order to the model to balance better?

Should I change the loss function so it penalizes more high values of observed?

Thank you in advance,

Francis

**Comment by hetong007** [comment]
IMO, changing the loss function might be a direct solution if doing it correctly. By saying "doing it correctly" I mean that you know what you actually want to control and you can express it as a formula in the loss function.

For instance, if you care more about the difference of the sum, *probably* using the absolute error as loss is better than the default squared loss. This is just an example, as the distribution of your target should also be taken into account: are most of them low values, any super high spike in the distribution, etc.

**Comment by carpenn** [comment]
Francis,

Not quite sure the nature of the data you are using but have you looked through the Allstate claims severity Kaggle competition? (https://www.kaggle.com/c/allstate-claims-severity)

That's an insurance claim severity modelling problem in which competitors used xgboost without issue. 

I've not experienced issues with xgboost on claim severity though as hetong007 says you need to carefully consider which loss function to use as that will affect whether xgboost replicates the population mean or median, which for insurance severity distributions can be very different.

**Comment by khotilov** [comment]
@Frankyboy1994 You've missed the important detail about what exactly is the loss function that you are using. For many kinds of losses you should not expect them to have such a "balance" property; and to model a sum you might need to use a different loss than for the modeling of individual claims.
